src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,278 @@
+/*-------------------------------------------------------------------------
+ *
+ * role.c
+ *    Commands for ALTER ROLE statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#if PG_VERSION_NUM >= 120000
+#include "access/table.h"
+#endif
+#include "catalog/catalog.h"
+#include "catalog/pg_authid.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/deparser.h"
+#include "distributed/master_protocol.h"
+#include "distributed/worker_transaction.h"
+#include "nodes/makefuncs.h"
+#include "nodes/parsenodes.h"
+#include "utils/acl.h"
+#include "utils/builtins.h"
+#include "utils/rel.h"
+#include "utils/syscache.h"
+
+static const char * ExtractEncryptedPassword(Oid roleOid);
+static const char * CreateAlterRoleIfExistsCommand(AlterRoleStmt *stmt);
+
+/* controlled via GUC */
+bool EnableAlterRolePropagation = false;
+
+/*
+ * ProcessAlterRoleStmt actually creates the plan we need to execute for alter
+ * role statement.
```

Comment:
Typically, the `Plan...` corresponding of the statement creates the tasks to execute. However, for some specific cases -- like this -- we do it on `Process...` part.  So, maybe add the reason for that to the comment (which is because we need to read the encrypted password from pg_authid, and that is true after `standardProcessUtility` is called)